.pre-commit-config.yaml

```diff
@@ -0,0 +1,44 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+      - id: check-yaml
+        args: [--allow-multiple-documents]
+      - id: check-json
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/pycqa/isort
+    rev: 5.11.5
+    hooks:
+      - id: isort
+        name: isort
+        entry: isort --profile google
+  - repo: https://github.com/psf/black
+    rev: 24.2.0
+    hooks:
+    - id: black
+      files: sdk/.*
```

Comment:
Should we also format files in examples directory ? 
@kubeflow/wg-training-leads do we have any other location where we store Python files ?

Reply:
Added the `examples` dir to this hook in a standalone [commit](https://github.com/kubeflow/training-operator/pull/2184/commits/1b35e252e3ceebf83d3699f02965ddc40f850439) so that you can review the scope of the changes. I'm in favor of it FWIW. Consistency across the whole codebase ftw.